### PR TITLE
Add ipaas-verifier + consolidation

### PR DESCRIPTION
**Pull request by rhuss**
This the initial shot to the templates for adding a DC and Service for the ipaas-verifier. Works nicely, but would love to have some quick review on this before I add them to the other templates.

### CODE REVIEW

**Comment by rhuss** [comment]
This PR now combines the following changes, adapted for template usage

* Quite some enhancement of the README (include the information from #28)
* Included #27 and #21
* Reorganized directories for a clearer structure: Top-level files are generated templates + README nly. Template generation --> generate/ . Supporting scripts and descriptors --> support/
* Moved "single-tenant" to "restricted". hope this its ok, but its more consistent in the naming to have only a single part name instead of two parts, which do not mix up well in combined names like for the generated templates.
* Removed "redhat-" prefix from template names. If this intended 'brandwise', then I can reinsert it. But removing the prefix makes the already long names a bit shorter. 

@jimmidyson @chirino is the restructuring ok so ? Happy to discuss and adapt it.

@dsimansk @jludvice Could you please have a quick look whether I haven't forgotten anything from your PRs ?

**Comment by rhuss** [comment]
For an impression of the new directory structure, please have a look at https://github.com/rhuss/openshift-templates/tree/pr/ipaas-verifier

**Comment by rhuss** [comment]
@iocanel forgot to ping you. Included and renamed the ephemeral template, too.

**Review by jimmidyson — APPROVED** [review]

**Comment by rhuss** [comment]
Just verified the template for Minishift and it seems to work basically fine.

There was an interesting issue with the DB template, I'd like to share:

The template had an error because of a new line in the readiness check with psql. Strangely that was not a YAML syntax error. So the DB didn't came up properly and the first pod was killed. When second pod started it had some random problems because it got back the same bound PVC and hence a (corrupted) db file. This worked sometimes, sometimes not. So I conclude plain postgresql is not so well suited for OpenShift even when one could afford a downtime.

**Comment by rhuss** [comment]
Hey @jimmidyson and my new friend @ghost : What's the process when an "approved" label was attached ? Could the PR then merged or should we wait for more approvals ? What the benefit over the normal Github review approval workflow ?

File: README.md
Comment (jludvice) [inline README.md]: Hi @rhuss - sorry for nitpicking but it looks like there is some typo which breaks table formatting.
screenshot_20170405_093728

Comment (rhuss) [reply README.md]: @jludvice you are welcome, no nitpicking at all !

Comment (rhuss) [reply README.md]: fixed

**Comment by jimmidyson** [comment]
@rhuss I'm hacking a bot that will do automerge once a PR is reviewed (labeled as `approved` - integrations can't view PR reviews atm for some reason...) and also all required status checks have passed. Should have it finished today and applied to all our repos. Sure it will need tweaking but certainly simplifies workflow...

**Comment by jimmidyson** [comment]
As for @ghost, that's because I had to recreate the integration which effectively deletes the user and creates a new one... Ask github why :-b

**Comment by rhuss** [comment]
@jimmidyson cool ;-) where can I find the source of your bot ?

Merging this pr now, we can tune it later further.

**Pull request merged**
